src/lightning/fabric/utilities/distributed.py

```diff
@@ -23,9 +27,67 @@ class group:  # type: ignore
         WORLD = None
 
 
+if TYPE_CHECKING:
+    from lightning.fabric.strategies import Strategy
+
+
 log = logging.getLogger(__name__)
 
 
+def is_shared_filesystem(strategy: "Strategy", path: Optional[_PATH] = None, timeout: int = 3) -> bool:
+    """Checks whether the filesystem under the given path is shared across all processes.
+
+    This function should only be used in a context where distributed is initialized.
+
+    Args:
+        strategy: The strategy being used, either from Fabric (``fabric.strategy``) or from Trainer
+            (``trainer.strategy``).
+        path: The path to check. Defaults to the current working directory. The user must have permissions to write
+            to this path or the parent folder, and the filesystem must be writable.
```

Comment:
edge case: what if the user passes a filepath where the parent folder doesn't exist? Should we traverse up until we find an existing path? Should it error out instead?

Reply:
For simplicity, I suggest making an error. Because otherwise, ultimately you would always reach the root filesystem as a parent path that exists but it is unlikely that the user would have write access. So in most cases it would be problematic if we did that upward traversal.